src/Type/Definition/NamedTypeImplementation.php

```diff
@@ -44,7 +45,7 @@ protected function inferName(): string
     /** @throws InvariantViolation */
     public function isBuiltInType(): bool
     {
-        return \array_key_exists($this->name, Type::builtInTypes());
+        return \in_array($this->name, [...Type::STANDARD_TYPE_NAMES, ...Introspection::TYPE_NAMES], true);
```

Comment:
Let's optimize further and extract the array to a `const BUILT_IN_TYPE_NAMES`, maybe in `Type`?

Reply:
On it.